cpp/src/unary/cast_ops.cu

```diff
@@ -146,14 +150,41 @@ constexpr inline auto is_supported_cast()
   return is_supported_non_fixed_point_cast<From, To>() || is_supported_fixed_point_cast<From, To>();
 }
 
-template <typename SourceT>
+template <typename From, typename To>
+struct device_cast {
+  __device__ To operator()(From element) { return static_cast<To>(element); }
+};
+
+template <typename T, typename std::enable_if_t<is_fixed_point<T>()>* = nullptr>
```

Comment:
This function needs docs.  Definition of "rescale" is unclear to the outside reader.

Reply:
Added (and cleaned up)